docs/source/integrations/building-integrations.md

```diff
@@ -1,54 +1,363 @@
+along to the user's `context` function.
 
-// 3. Set up our middleware for our framework specific server.
-app.use(cors(), bodyParser.json(), expressMiddleware(server));
+Apollo Server exports the `ContextFunction` type, which is a generic type that
+is useful for integrations in defining their API. The first type argument
+defines the arguments that will be passed to your user's `context` function. The
+Express integration uses it above in the `ExpressMiddlewareOptions` interface so
+that users get a strongly typed `context` function with correct parameter
+typings. The second type argument defines the return type of the user's
+`context` function, and should be the same `TContext` generic as used in the
+`ApolloServer` instance.
 
-// 4. Start the framework specific server
-app.listen({ port: 4000 }, () => {
-  console.log(`🚀 Server ready at http://localhost:4000`);
-});
+```ts
+interface ExpressContextFunctionArgument {
+  req: express.Request;
+  res: express.Response;
+}
+
+const defaultContext: ContextFunction<
+  [ExpressContextFunctionArgument],
+  any
+> = async () => ({});
+
+const context: ContextFunction<[ExpressContextFunctionArgument], TContext> =
+  options?.context ?? defaultContext;
 ```
 
-Note that your integration is responsible for setting up [body-parser](https://www.npmjs.com/package/body-parser) and [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) configuration.
+### Handling Requests
+
+This section is where implementations can expect to diverge from the Express
+implementation the most. The request handler has 4 main functions:
+1. Ensure the request is valid
+2. Construct an `HTTPGraphQLRequest` object from the incoming request
+3. Execute the GraphQL request via Apollo Server
+4. Return a well-formed response to the client
+
+
+#### Ensure the request is valid
+
+During this step, the Express handler checks for the existence of a `body` on
+the request. The Express handler expects the use of the `body-parser` package to
+parse the request body into a JavaScript object. We know that if there's no
+`body` on the request, the middleware isn't configured properly so we respond
+with an error.
+
 
+```ts
+if (!req.body) {
+  res.status(500);
+  res.send(
+    '`req.body` is not set; this probably means you forgot to set up the ' +
+      '`body-parser` middleware before the Apollo Server middleware.',
+  );
+  return;
+}
+```
+
+Because the Express implementation uses the `body-parser` package, all requests
+with a `content-type: application/json` header will either be parsed into a
+JavaScript object or an error will be returned before the handler is even
+called. It's an integration's responsibility to ensure that the body is parsed
+appropriately based on the `content-type` header and that errors are returned
+for invalid requests.
+
+FIXME: add note about qs and get query string parameter parsing
+
+Apollo Server expects a few "types" of requests:
```

Comment:
I don't know that this list is actually helpful for this particular page. The integration code doesn't have to care about any of this as long as it can pass on the full body and set of query parameters.

Reply:
Yeah this is the section I knew I'd redo,

It can basically be simplified to: "these are the shapes you need to parse your body and query string params into and AS handles the rest". Does that sound about right?

Reply:
I guess my point is that this section doesn't even need to contain words like `query` and `operationName`; I'm not sure from your question if that's what you're proposing or not.